src/extension.ts

```diff
@@ -108,14 +110,13 @@ function startServer(context: ExtensionContext, javaHome: string) {
 
   context.subscriptions.push(
     commands.registerCommand("metals.restartServer", async () => {
-      const serverPid = client["_serverProcess"].pid;
-      await exec(`kill ${serverPid}`);
+      await client.sendRequest(ShutdownRequest.type);
+      client.sendNotification(ExitNotification.type);
```

Comment:
I think we should always fallback to killing the pid after a timeout, to account for completely unresponsive servers.

Also, have you tested this? This previously relied on `language-client` to restore the server when killed, I'm not sure it will restart it if terminated gracefully.

Reply:
Good catch. It now tries to gracefully shutdown but if there's no response in 4s then we kill the process. I manually tested the `kill` path with a local version of Metals that sleeps during `shutdown`. Would be good to get a second eye on the timeout logic though, easy to mess this up.